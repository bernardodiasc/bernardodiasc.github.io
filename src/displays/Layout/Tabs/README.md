# Tabs

## How to use

```
import Tabs from 'displays/Layout/Tabs'
```

```
<Tabs />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
