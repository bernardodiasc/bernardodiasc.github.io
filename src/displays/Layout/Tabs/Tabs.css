.Layout-Tabs {
  font-size: var(--text-size);
  display: flex;
}

.Layout-Tabs__tab {
  display: inline-block;
  background: var(--content-background);
  margin: 0 0 0 5px;
  border-top-right-radius: 5px;
  border-top-left-radius: 5px;
}

.Layout-Tabs__tab:first-of-type {
  margin: 0;
}

.Layout-Tabs__link {
  display: block;
  width: 18px;
  height: 18px;
  color: var(--meta-color);
  transition: color 0.3s ease;
  padding: 0.3em 0.8em;
  box-sizing: content-box;
  box-shadow: 0 -1px 3px 0 rgba(45, 45, 45, 0.4) inset;
}

.Layout-Tabs__link:hover {
  color: var(--text-color);
}

.Layout-Tabs__link--active {
  color: var(--heading-color);
  box-shadow: none;
}
