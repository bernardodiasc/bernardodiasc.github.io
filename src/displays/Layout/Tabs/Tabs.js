import React, { PureComponent } from 'react'
import './Tabs.css'

import AppLink from 'displays/AppLink'
import Icon from 'displays/Icon'

class Tabs extends PureComponent {
  static defaultProps = {}

  render() {
    return (
      <ul className="Layout-Tabs">
        <li className="Layout-Tabs__tab">
          <AppLink
            to="/"
            title="Latest posts"
            className="Layout-Tabs__link"
            activeClassName="Layout-Tabs__link--active"
          >
            <Icon icon="home" />
          </AppLink>
        </li>
        <li className="Layout-Tabs__tab">
          <AppLink
            to="search"
            title="Search for content"
            className="Layout-Tabs__link"
            activeClassName="Layout-Tabs__link--active"
          >
            <Icon icon="explore" />
          </AppLink>
        </li>
        <li className="Layout-Tabs__tab">
          <AppLink
            to="archive"
            title="Content archive"
            className="Layout-Tabs__link"
            activeClassName="Layout-Tabs__link--active"
          >
            <Icon icon="calendar" />
          </AppLink>
        </li>
        <li className="Layout-Tabs__tab">
          <AppLink
            to="about-me"
            title="About me"
            className="Layout-Tabs__link"
            activeClassName="Layout-Tabs__link--active"
          >
            <Icon icon="guy-fawkes-mask" />
          </AppLink>
        </li>
      </ul>
    )
  }
}

export default Tabs
