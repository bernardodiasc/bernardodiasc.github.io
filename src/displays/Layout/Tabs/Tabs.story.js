import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import Tabs from './Tabs'
import TabsReadme from './README.md'
storiesOf('Displays/Layout/Tabs', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{TabsReadme}</Docs>)
  .add('Example', () => <Tabs />)
  .add('Minimal render', () => <Tabs />)
