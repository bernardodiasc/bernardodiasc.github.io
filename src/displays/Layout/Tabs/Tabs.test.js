import React from 'react'
import { shallow } from 'enzyme'

import Tabs from './Tabs.js'

it('renders without crashing', () => {
  const component = shallow(<Tabs />)
  expect(component).toMatchSnapshot()
})
