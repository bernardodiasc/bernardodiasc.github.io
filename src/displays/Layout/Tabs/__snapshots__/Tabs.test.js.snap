// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ul
  className="Layout-Tabs"
>
  <li
    className="Layout-Tabs__tab"
  >
    <AppLink
      activeClassName="Layout-Tabs__link--active"
      className="Layout-Tabs__link"
      title="Latest posts"
      to="/"
    >
      <Icon
        icon="home"
      />
    </AppLink>
  </li>
  <li
    className="Layout-Tabs__tab"
  >
    <AppLink
      activeClassName="Layout-Tabs__link--active"
      className="Layout-Tabs__link"
      title="Search for content"
      to="search"
    >
      <Icon
        icon="explore"
      />
    </AppLink>
  </li>
  <li
    className="Layout-Tabs__tab"
  >
    <AppLink
      activeClassName="Layout-Tabs__link--active"
      className="Layout-Tabs__link"
      title="Content archive"
      to="archive"
    >
      <Icon
        icon="calendar"
      />
    </AppLink>
  </li>
  <li
    className="Layout-Tabs__tab"
  >
    <AppLink
      activeClassName="Layout-Tabs__link--active"
      className="Layout-Tabs__link"
      title="About me"
      to="about-me"
    >
      <Icon
        icon="guy-fawkes-mask"
      />
    </AppLink>
  </li>
</ul>
`;
